tests/pr_test.go

```diff
@@ -598,3 +598,116 @@ func TestRunVsiExtention(t *testing.T) {
 		logger.Log(t, "END: Destroy (existing resources)")
 	}
 }
+
+func TestRunUpgradeVsiExtention(t *testing.T) {
```

Comment:
Is alot of the code here duplicated from the `TestRunVsiExtention` test? Can we maybe remove duplication by adding the common code to a function and having each test call that function?